### PR TITLE
Moxy with base

### PR DESCRIPTION
**Pull request by MorglenRazor**
Create Base structure and create showMessage method

### CODE REVIEW

File: app/src/main/java/me/morglenrazor/imdbcollector/presentaiton/base/BaseView.kt [hunk 1]
```diff
@@ -1,9 +1,12 @@
 package me.morglenrazor.imdbcollector.presentaiton.base
 
+import android.view.View
 import moxy.MvpView
+import moxy.viewstate.strategy.StateStrategyType
 import moxy.viewstate.strategy.alias.AddToEndSingle
+import java.lang.Error
 
 interface BaseView : MvpView {
     @AddToEndSingle
```
Comment (alexandervortex) [inline app/src/main/java/me/morglenrazor/imdbcollector/presentaiton/base/BaseView.kt]: If you show an error, and then rotate your phone - error shows again. U need another annotation

File: app/src/main/java/me/morglenrazor/imdbcollector/ui/base/BaseActivity.kt [hunk 2]
```diff
@@ -1,30 +1,23 @@
...
 import moxy.presenter.InjectPresenter
 
-class BaseActivity : MvpAppCompatActivity(), BaseView {
-
-    @InjectPresenter
-    lateinit var presenter: BasePresenter
-
+abstract class BaseActivity : MvpAppCompatActivity(), BaseView {
     override fun onCreate(savedInstanceState: Bundle?) {
-
```
Comment (alexandervortex) [inline app/src/main/java/me/morglenrazor/imdbcollector/ui/base/BaseActivity.kt:17]: U can remove onCreate overriding.

File: app/src/main/java/me/morglenrazor/imdbcollector/ui/base/BaseActivity.kt [hunk 3]
```diff
@@ -1,30 +1,23 @@
...
-            presenter.onButtonClicked(mainEditText.text.toString())
-        }
 
     }
 
-    override fun setText(text: String){
-        mainTextView.text = text
+    override fun showMessage(msg: String) {
+        val toast = Toast.makeText(this, msg, Toast.LENGTH_LONG)
+        toast.show()
```
Comment (alexandervortex) [inline app/src/main/java/me/morglenrazor/imdbcollector/ui/base/BaseActivity.kt]: U can do it with Toast.makeText(this, msg, Toast.LENGTH_LONG).show()

File: app/src/main/java/me/morglenrazor/imdbcollector/ui/base/BaseFragment.kt [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+
+import android.widget.Toast
+import kotlinx.android.synthetic.main.activity_main.*
+import me.morglenrazor.imdbcollector.presentaiton.base.BaseView
+import moxy.MvpAppCompatFragment
+
+abstract class BaseFragment : MvpAppCompatFragment(), BaseView{
+
+    override fun showMessage(msg: String) {
+            val toast = Toast.makeText(activity, msg, Toast.LENGTH_LONG)
```
Comment (alexandervortex) [inline app/src/main/java/me/morglenrazor/imdbcollector/ui/base/BaseFragment.kt]: U can remove showMessage from your Activity. If you have that method at Fragment. U should use only one method.

File: app/src/main/java/me/morglenrazor/imdbcollector/ui/main/MainActivity.kt [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        setContentView(R.layout.activity_main)
+
+        mainButtonShowTxt.setOnClickListener {
+            val txt = mainEditText.text.toString()
+            presenter.onButtonClicked(txt)
+//            val toast = Toast.makeText(applicationContext, "Ghbdtn", Toast.LENGTH_LONG)
+//            toast.show()
```
Comment (alexandervortex) [inline app/src/main/java/me/morglenrazor/imdbcollector/ui/main/MainActivity.kt]: Always remove your comments and another unnecessary code before creating PR

File: app/src/main/java/me/morglenrazor/imdbcollector/presentaiton/base/BaseView.kt (hunk 1 above)
Comment (MorglenRazor) [reply app/src/main/java/me/morglenrazor/imdbcollector/presentaiton/base/BaseView.kt]: Alright, but i just forgot delete this import

**Pull request merged**